src/rgw/rgw_main.cc

```diff
@@ -342,18 +342,51 @@ int main(int argc, const char **argv)
   }
 
   // S3 website mode is a specialization of S3
-  bool s3website_enabled = apis_map.count("s3website") > 0;
-  if (apis_map.count("s3") > 0 || s3website_enabled)
-    rest.register_default_mgr(set_logging(new RGWRESTMgr_S3(s3website_enabled)));
+  const bool s3website_enabled = apis_map.count("s3website") > 0;
+  // Swift API entrypoint could placed in the root instead of S3
+  const bool swift_at_root = g_conf->rgw_swift_url_prefix == "/";
```

Comment:
I don't think so. The purpose of the `swift_at_root` is to indicate whether someone _requested_ putting Swift API at root. It could be placed there only if Swift API is enabled and S3 (and S3 website) are turned off.